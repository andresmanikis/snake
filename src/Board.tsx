import styles from "./Board.module.css";

interface Props {
  stateMatrix: Array<Array<boolean>>;
}

export function Board({ stateMatrix }: Props) {
  return (
    <table className={styles.table}>
      <tbody>
        {stateMatrix.map((row, index) => (
          <tr key={index} className={styles.tr}>
            {row.map((col, index) => {
              let classes = styles.td;

              if (col === true) {
                classes += " " + styles.active;
              }

              return <td key={index} className={classes}></td>;
            })}
          </tr>
        ))}
      </tbody>
    </table>
  );
}
