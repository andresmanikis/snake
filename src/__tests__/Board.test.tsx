import { render } from "@testing-library/react";

import { Board } from "../Board";

it("renders a state matrix", () => {
  // prettier-ignore
  const stateMatrix = [
    [false, false, false, false, false],
    [false, true,  false, true,  false],
    [false, false, true,  false, false],
    [false, true,  false, true,  false],
    [false, false, false, false, false],
  ];

  const { container } = render(<Board stateMatrix={stateMatrix} />);
  expect(container).toMatchSnapshot();
});
