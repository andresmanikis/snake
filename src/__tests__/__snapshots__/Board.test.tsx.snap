// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a state matrix 1`] = `
<div>
  <table
    class="table"
  >
    <tbody>
      <tr
        class="tr"
      >
        <td
          class="td"
        />
        <td
          class="td"
        />
        <td
          class="td"
        />
        <td
          class="td"
        />
        <td
          class="td"
        />
      </tr>
      <tr
        class="tr"
      >
        <td
          class="td"
        />
        <td
          class="td active"
        />
        <td
          class="td"
        />
        <td
          class="td active"
        />
        <td
          class="td"
        />
      </tr>
      <tr
        class="tr"
      >
        <td
          class="td"
        />
        <td
          class="td"
        />
        <td
          class="td active"
        />
        <td
          class="td"
        />
        <td
          class="td"
        />
      </tr>
      <tr
        class="tr"
      >
        <td
          class="td"
        />
        <td
          class="td active"
        />
        <td
          class="td"
        />
        <td
          class="td active"
        />
        <td
          class="td"
        />
      </tr>
      <tr
        class="tr"
      >
        <td
          class="td"
        />
        <td
          class="td"
        />
        <td
          class="td"
        />
        <td
          class="td"
        />
        <td
          class="td"
        />
      </tr>
    </tbody>
  </table>
</div>
`;
